core/src/com/unciv/logic/automation/civilization/DiplomacyAutomation.kt

```diff
@@ -274,7 +276,9 @@ object DiplomacyAutomation {
 
         val enemiesCiv = civInfo.diplomacy.filter { it.value.diplomaticStatus == DiplomaticStatus.War }
             .map { it.value.otherCiv() }
-            .filterNot { it == civInfo || it.isBarbarian() || it.cities.isEmpty() }
+            .filterNot { it == civInfo || it.isBarbarian() || it.cities.isEmpty() 
+                    || it.getDiplomacyManager(civInfo).hasFlag(DiplomacyFlags.DeclaredWar) 
+                    || civInfo.getDiplomacyManager(it).hasFlag(DiplomacyFlags.DeclaredWar) }
```

Comment:
We might want to add a method canDeclareWar() in diplomacyManager to simplify this.
We might also want to add a doesEitherSideHaveFlag() method as well.